### PR TITLE
feat(STONE-841): remove Environment field from ITS CRD

### PR DESCRIPTION
**Pull request by jsztuka**
## Maintainers will complete the following section

- [ ] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/redhat-appstudio/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/redhat-appstudio/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

As part of stopping provisioning ephemeral environments, we need to update our IntegrationTestScenario by removing **environment** field.

- Add webhook that will convert v1beta1 version to v1beta2
- Update unit test affected by this change
- Remove all the code that will become redundat after applying this change
note: second and third point will be substracted from this PR after [This PR](https://github.com/redhat-appstudio/integration-service/pull/572) gets merged

### CODE REVIEW

File: controllers/snapshot/snapshot_adapter_test.go
Comment (jsztuka) [inline controllers/snapshot/snapshot_adapter_test.go:1000]: This line will be updated once the https://github.com/redhat-appstudio/integration-service/pull/572/
will see light of day.

File: api/v1alpha1/integrationtestscenario_conversion.go
Comment (MartinBasti) [inline api/v1alpha1/integrationtestscenario_conversion.go:98]: so we don't need this conversion anymore?

File: api/v1beta2/groupversion_info.go
Comment (MartinBasti) [inline api/v1beta2/groupversion_info.go:2]: 2024

File: config/crd/patches/webhook_in_integrationtestscenarios.yaml
Comment (MartinBasti) [inline config/crd/patches/webhook_in_integrationtestscenarios.yaml:12]: so we don't want to have conversions from v1alpha1 anymore?

File: controllers/scenario/scenario_adapter.go
Comment (MartinBasti) [inline controllers/scenario/scenario_adapter.go:92]: Are you sure that ContinueProcessing should be here?

**Comment by MartinBasti** [comment]
This PR duplicates a lot of work done in https://github.com/redhat-appstudio/integration-service/pull/572 why we don't merge 572 first and then just rebase, instead of reviewing the same code change twice?

File: PROJECT
Comment (MartinBasti) [reply PROJECT:20]: would be amazing to keep changes isolated, and when request is to remove filed and make new API, just remove field, not whole old API.

Can you gurantee that this change will working prod? IIRC we had issue with removing old API before and we had to keep it and only mark it as old API.

File: api/v1beta2/integrationtestscenario_types.go
Comment (MartinBasti) [inline api/v1beta2/integrationtestscenario_types.go:2]: 2024

**Comment by jsztuka** [comment]
Changes were tested following way:
1. Deployed integration-service latest version by infra-deployments.
2. Created v1beta1 integrationtest scenario with environment specified.
3. Deployed changes in this PR on the same cluster
4. Checked and confirmed the format of migrated v1beta2 scenario.
Example of migrated scenario: 
Screenshot from 2024-04-23 20-19-46

File: controllers/snapshot/snapshot_adapter_test.go
Comment (jsztuka) [reply controllers/snapshot/snapshot_adapter_test.go:1000]: Its set to false in main, ignore previous comment.

File: controllers/snapshot/snapshot_adapter_test.go
Comment (hongweiliu17) [inline controllers/snapshot/snapshot_adapter_test.go:64]: Since we won't have `environment` in ITS, then we may update the name and test for ITS with or without environment to make it clean, how do you think?

Comment (jsztuka) [reply controllers/snapshot/snapshot_adapter_test.go:64]: Yes I was thinking of keeping just integrationTestScenario and removing the withoutEnv one.

Comment (jsztuka) [reply controllers/snapshot/snapshot_adapter_test.go]: Updated so we keep only integrationTestScenario.

File: controllers/scenario/scenario_controller_test.go
Comment (MartinBasti) [inline controllers/scenario/scenario_controller_test.go:210]: why `failScenario` has been removed?

Comment (jsztuka) [reply controllers/scenario/scenario_controller_test.go:210]: Sorry, added back, wierd that it had no effect on passing the tests.

File: controllers/scenario/scenario_controller_test.go
Comment (jsztuka) [reply controllers/scenario/scenario_controller_test.go:179]: So it turned out that just before we initiate reconciler, we would need to delete scenarios. In this case on 2 places.

Comment (hongweiliu17) [reply controllers/scenario/scenario_controller_test.go:179]: But if we delete it before running reconciler, so how can we make sure it is reconciled according to code https://github.com/redhat-appstudio/integration-service/blob/main/controllers/scenario/scenario_controller.go#L67-L69

Comment (jsztuka) [reply controllers/scenario/scenario_controller_test.go:179]: I see, we tend to delete the resource before reconcile because of the content of ctrl.Request. We have 2 options of dealing with this.
1. Deleting the resource as we do on many places
2. Rename the Name of the object that should be processed by reconcile.

Comment (jsztuka) [reply controllers/scenario/scenario_controller_test.go:179]: I say we stick with the deletion before reconcile, wdyt?

Comment (MartinBasti) [reply controllers/scenario/scenario_controller_test.go:179]: I really don't understand how this is supposed to fix test. 
Firstly in BeforeEach we create the scenario and then it's just removed before the useful part of test starts?

File: controllers/scenario/scenario_adapter.go
Comment (dirgim) [inline controllers/scenario/scenario_adapter.go:152]: Is this removal necessary for this PR? If not, it can be handled as part of [STONEINTG-815](https://issues.redhat.com/browse/STONEINTG-815).

File: controllers/scenario/scenario_adapter.go
Comment (dirgim) [inline controllers/scenario/scenario_adapter.go:173]: Is this removal necessary for this PR? If not, it can be handled as part of [STONEINTG-815](https://issues.redhat.com/browse/STONEINTG-815).

File: controllers/scenario/scenario_controller_test.go
Comment (dirgim) [reply controllers/scenario/scenario_controller_test.go:179]: Would setting the `deletionTimestamp` on the scenario be enough to make this work? That seems preferrable to deleting the resource.

On the other hand, maybe this shouldn't need to be handled as part of this PR as I mentioned in [this comment](https://github.com/redhat-appstudio/integration-service/pull/666/files#r1579168540).

File: controllers/scenario/scenario_adapter.go
Comment (jsztuka) [reply controllers/scenario/scenario_adapter.go:173]: Let me switch it back, you re right, it should be part of 815

File: controllers/scenario/scenario_controller_test.go
Comment (jsztuka) [reply controllers/scenario/scenario_controller_test.go]: Thanks to Kruno for finding the issue, we needed to set the tests to be ordered. Finalizers were taking too much of a time between those deletions, causing the errors.

File: api/v1beta2/integrationtestscenario_conversion.go
Comment (hongweiliu17) [inline api/v1beta2/integrationtestscenario_conversion.go:2]: 2024

**Comment by hongweiliu17** [comment]
No comment from me. Will approve when all conversion are closed. Thanks!

File: PROJECT
Comment (jsztuka) [reply PROJECT]: All versions included in PROJECT.

**Review by MartinBasti — APPROVED** [review]

**Review by dirgim — APPROVED** [review]
LGTM

**Review by hongweiliu17 — APPROVED** [review]
LGTM

**Pull request merged**
